core/src/main/java/org/frankframework/receivers/Receiver.java

```diff
@@ -1693,9 +1754,53 @@ public void increaseRetryIntervalAndWait(Throwable t, String description) {
 				throwEvent(RCV_SUSPENDED_MONITOR_EVENT);
 			}
 		}
+		suspendReceiver(currentInterval);
+	}
+
+	/**
+	 * Get the transaction timeout in seconds for this receiver, or 0 if it is not possible to determine this.
+	 *
+	 * @return The transaction timeout in seconds configured on the Receiver, or from the TransactionManager if it was not
+	 * configured on the receiver.
+	 */
+	private int getActualTransactionTimeout() {
+		if (getTransactionTimeout() != 0) {
+			return getTransactionTimeout();
+		}
+		if (txManager instanceof AbstractPlatformTransactionManager platformTransactionManager) {
+			return platformTransactionManager.getDefaultTimeout();
+		}
+		return AppConstants.getInstance(configurationClassLoader).getInt("transactionmanager.defaultTransactionTimeout", 0);
+	}
+
+	/**
+	 * Derive the maximum retry delay to use for this receiver, based on the configured max retry
+	 * delay for the receiver, the global default max retry delay, and the transaction timeout.
+	 * The actual maximum retry delay can never be more than half the {@link #getActualTransactionTimeout()}.
+	 *
+	 * @return Maximum retry delay in seconds.
+	 */
+	private int getActualMaxRetryDelay() {
+		int actualMaxRetryDelay;
+		int actualTransactionTimeout = getActualTransactionTimeout();
+		if (isTransacted() && actualTransactionTimeout > 0) {
+			actualMaxRetryDelay = min(maxRetryDelay, actualTransactionTimeout / 2);
+			log.debug("{} Max retry delay set to {} seconds to avoid automatic transaction timeout due to delay", this::getLogPrefix, ()->actualMaxRetryDelay);
```

Comment:
Misschien dat het handiger is een configuratie warning te geven (in configure) wanneer de `maxRetryDelay` groter is dan 2x `platformTransactionManager.getDefaultTimeout() / getTransactionTimeout()`?